### PR TITLE
library

### PR DESCRIPTION
**Pull request by hung070996**
https://edu-redmine.framgia.vn/issues/1154
https://edu-redmine.framgia.vn/issues/1155
https://edu-redmine.framgia.vn/issues/1157
https://edu-redmine.framgia.vn/issues/1158
https://edu-redmine.framgia.vn/issues/1159
https://edu-redmine.framgia.vn/issues/1160
https://edu-redmine.framgia.vn/issues/1161
https://edu-redmine.framgia.vn/issues/1162

### CODE REVIEW

File: MyEditor/MyEditor/Repositories/LibraryRepository.swift [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                    guard let album = asset as? PHAssetCollection else {
+                        return
+                    }
+                    let options = PHFetchOptions()
+                    options.predicate = NSPredicate(format: "mediaType = %d", PHAssetMediaType.image.rawValue)
+                    let assets = PHAsset.fetchAssets(in: album, options: options)
+                    guard let _ = assets.firstObject else {
+                        return
+                    }
+                    PHImageManager.default().requestImage(for: assets[assets.count - 1], targetSize: CGSize(width: Constant.defaultSize, height: Constant.defaultSize), contentMode: PHImageContentMode.default, options: nil, resultHandler: { (result, info) in
```
Comment (tgdong2296) [inline MyEditor/MyEditor/Repositories/LibraryRepository.swift]: dòng này dài quá thì break line nhé

File: MyEditor/MyEditor/Scene/Library/Cell/ListAlbumCell.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+import UIKit
+import Reusable
+
+class ListAlbumCell: UITableViewCell, NibReusable {
+
+    @IBOutlet private var amountLabel: UILabel!
+    @IBOutlet private var nameLabel: UILabel!
+    @IBOutlet private var iconAlbum: UIImageView!
+    override func awakeFromNib() {
```
Comment (tgdong2296) [inline MyEditor/MyEditor/Scene/Library/Cell/ListAlbumCell.swift]: thêm break line

File: MyEditor/MyEditor/Scene/Library/Cell/ListAlbumCell.swift [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+//
+
+import UIKit
+import Reusable
+
+class ListAlbumCell: UITableViewCell, NibReusable {
+
+    @IBOutlet private var amountLabel: UILabel!
+    @IBOutlet private var nameLabel: UILabel!
+    @IBOutlet private var iconAlbum: UIImageView!
```
Comment (tuan188) [inline MyEditor/MyEditor/Scene/Library/Cell/ListAlbumCell.swift]: iconAlbumImageView

File: MyEditor/MyEditor/Scene/Library/LibraryNavigator.swift [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+//
+//  Created by Do Hung on 8/31/18.
+//  Copyright © 2018 Do Hung. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+protocol LibraryNavigatorType {
+    func toListPhotoInAlbum(album: Album)
```
Comment (tuan188) [inline MyEditor/MyEditor/Scene/Library/LibraryNavigator.swift]: func toListPhotoInAlbum(_ album: Album)

viết vậy lúc gọi đỡ bị trùng lặp nhiều chữ album

File: MyEditor/MyEditor/Scene/Library/LibraryNavigator.swift [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+struct LibraryNavigator: LibraryNavigatorType {
+    unowned let navigationController: UINavigationController
+    
+    func toListPhotoInAlbum(album: Album) {
```
Comment (tuan188) [inline MyEditor/MyEditor/Scene/Library/LibraryNavigator.swift]: toListPhotoInAlbum(_ album: Album)

hoặc tốt hơn là 
toListPhoto(in album: Album)

File: MyEditor/MyEditor/Scene/Library/LibraryViewModel.swift [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    }
+    
+    let useCase: LibraryUseCaseType
+    let navigator: LibraryNavigator
+    
+    func transform(_ input: LibraryViewModel.Input) -> LibraryViewModel.Output {
+        let listAlbum = input.loadTrigger
+            .flatMapLatest { _ in
+                return self.useCase.getAlbums()
+                    .asDriverOnErrorJustComplete()
```
Comment (tgdong2296) [inline MyEditor/MyEditor/Scene/Library/LibraryViewModel.swift]: `.trackErrror(errorTracker)` để bắt các error bắn ra từ `Observable`

File: MyEditor/MyEditor/Scene/Library/LibraryNavigator.swift (hunk 4 above)
Comment (tuan188) [reply MyEditor/MyEditor/Scene/Library/LibraryNavigator.swift]: toListPhoto(in album: Album)

File: MyEditor/MyEditor/Scene/Library/LibraryViewModel.swift [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        }
+        return Output(
+            listAlbums: listAlbum.asDriver()
```
Comment (tuan188) [inline MyEditor/MyEditor/Scene/Library/LibraryViewModel.swift]: không cần asDriver()

**Pull request merged**